src/effects/backends/builtin/pitchshifteffect.cpp

```diff
@@ -160,6 +203,41 @@ void PitchShiftEffect::processChannel(
         SINT processedFrames = 0;
 
         while (processedFrames < blockFrames) {
+            // RubberBand is asking us for data. This practice could look
+            // like the "pull" implementation as known for RubberBand
+            // implementation, anyway, it is not for example
+            // for the following reasons:
+            //
+            // 1. Prefilling with zero samples to avoid dropouts.
+            // 2. When the last block of the input buffer is processed
+            //    and the amount of frames that RubberBand requires is greater
+            //    than the rest of the input buffer frames, only the rest
+            //    of the input buffer frames are provided and not the amount,
+            //    that RubberBand requires.
```

Comment:
Can RubberBand even use the rest, or is something bad happening. 

I expect the Rubberband users windowing functions. With 50 % overlap or such. If you now pass only a few samples the window is not full. What will happen? Rubberband fills is with old samples, CPU is increase / Rubberband does nothing, than there is no pint to even provide the samples / Rubberband pads the window with zeros which may create an audible glitch. (I have not checked so I might be wrong).

From this point of view a traditional pull model with an input buffer creating a fixed group delay seems to be more robust. This way we can always provide the expected samples for the desired output with no exceptions.

Reply:
rubberband actually skips the process call if we have provided not enough data: 
https://github.com/breakfastquay/rubberband/blob/10661f40c60038cef55c1a2d8e417dbca4a17616/src/faster/StretcherProcess.cpp#L335